### PR TITLE
fix: add api for java datastore/admin

### PR DESCRIPTION
**Pull request by dmitry-fa**
to:  @IlyaFaer, @mf2199, @AVaksman

32 changed files, 31 of them are generated. The only file changed is 'synth.py'

For java-datastore what I did:

0) installed python3 and [synthtool](https://github.com/googleapis/synthtool)
`   #> python3 -m pip install --user --upgrade git+https://github.com/googleapis/synthtool.git  `

1) forked the repository (java) and clone it
```
   #> git clone https://github.com/dmitry-fa/java-datastore.git
   #> cd java-datastore
```
2) ran verification test (to make sure I ran it correctly). For Java it looks like:
`   #> mvn verify `

3) updated existing synth.py in the root folder to run generation on 'datastore/admin' folder in addition to 'datastore' folder:
   See [files](https://github.com/dmitry-fa/java-datastore/pull/1/files) and search for `synth.py`

4) ran synthtool
`  #>  python3 -m synthtool`

5) saw what happened:
`   #> git status`

6) ran verification tests again
`   #> mvn verify `  (java specific)

7) prepared pull request

### CODE REVIEW

**Review by mf2199 — COMMENTED** [review]
IMHO, changes to the `synth.py` should be recorded but not published just yet. We were only asked to generate the code and make available. I'd leave `synth.py` as it was before and put the changes in the PR comments. This way they can always be added in a separate PR, if needed, while avoiding potential breakages.

**Comment by dmitry-fa** [comment]
> 
> 
> IMHO, changes to the `synth.py` should be recorded but not published just yet. We were only asked to generate the code and make available. I'd leave `synth.py` as it was before and put the changes in the PR comments. This way they can always be added in a separate PR, if needed, while avoiding potential breakages.

I think, synth.py is much important than generated sources. My guess, it's not needed to integrate sources they will be generated and updated by bots. Sources included in this PR just to demonstrate how the updated script works.

File: synth.py [hunk 1]
```diff
@@ -44,11 +44,26 @@
...
+protos = [
+  {
+    'service': 'datastore',
+    'version': 'v1',
+  },
+  {
+    'service': 'datastore-admin',
+    'version': 'v1',
+  }
+]
```
Comment (IlyaFaer) [inline synth.py:53]: I suppose you can exclude `version` field here and in line 60 just use a `"v1"` string, as they are similar in both services

**Review by IlyaFaer — COMMENTED** [review]
One comment, but overall looks okay: all the files are in `admin` directory, so it's gonna be the things that we have had to generate

File: synth.py (hunk 1 above)
Comment (dmitry-fa) [reply synth.py:53]: It's for the future use. They use to have an array of one element:  
`versions = ['v1'] ` to iterate over, so I would prefer to keep this.

**Pull request closed without merging**